torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -752,8 +755,10 @@ void Reducer::prepare_for_backward(
   // variables will be called, and we don't have to search the autograd graph
   // for presence of these hooks.
   if (outputs.empty()) {
+    find_unused_parameters_ = false;
```

Comment:
This does not guarantee to be correct either. What if one process's forward pass produces empty outputs but still sets `find_unused_parameters_ =rure`?

Reply:
Okay, will move it to the constructor and also address @zhaojuanmao 's comment.